BuildingBlocks/test/BuildingBlocks.Application.Tests/Mediatr/QuotaEnforcerBehaviorTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions;
+using Enmeshed.BuildingBlocks.Application.Attributes;
+using Enmeshed.BuildingBlocks.Application.MediatR;
+using Enmeshed.BuildingBlocks.Domain;
+using FluentAssertions;
+using MediatR;
+using Xunit;
+
+namespace Enmeshed.BuildingBlocks.Application.Tests.Mediatr;
+public class QuotaEnforcerBehaviorTests
+{
+    public QuotaEnforcerBehaviorTests() { }
+
+    [ApplyQuotasForMetrics("MetricKey")]
+    private class TestCommand : IRequest<IResponse> { }
+
+    private class IResponse { }
+
+    [Fact]
+    public async void Throws_QuotaExhaustedException_OnExhaustedQuota()
+    {
+        // Arrange
+        var metricStatusesStubRepository = new MetricStatusesStubRepository(new() {
+            new MetricStatus(new MetricKey("KeyOne"), DateTime.UtcNow.AddDays(1)) 
+        });
+
+        var userContextStub = new UserContextStub();
+        var behavior = new QuotaEnforcerBehavior<TestCommand, IResponse>(metricStatusesStubRepository, userContextStub);
+        var nextHasRan = false;
+
+        // Act
+        Func<Task> acting = async () => await behavior.Handle(
+             new TestCommand(),
+            () => {
+                nextHasRan = true;
+                return Task.FromResult(new IResponse());
+            },
+            CancellationToken.None);
+
+        // Assert
+        await acting.Should().ThrowAsync<QuotaExhaustedException>();
+        nextHasRan.Should().BeFalse();
```

Comment:
You're testing two things here. These should be separate tests.
The goal is to have a test only fail for exactly one reason (in this case: either because the exception is not thrown or because `next` wasn't called)